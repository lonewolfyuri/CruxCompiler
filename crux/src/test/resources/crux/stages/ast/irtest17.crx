// test local variable

func main() : void {
  var a : int;
  let a = 42;
  ::printInt(a);
}
