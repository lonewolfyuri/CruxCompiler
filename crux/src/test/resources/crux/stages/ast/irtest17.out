ast.DeclarationList(3,0)
  ast.FunctionDefinition(3,0)[Symbol(main), []]
    ast.StatementList(4,2)
      ast.VariableDeclaration(4,2)[Symbol(a)]
      ast.Assignment(5,2)
        ast.Name(5,6)[a]
        ast.LiteralInt(5,10)[42]
      ast.Call(6,2)[Symbol(printInt)]
        ast.Dereference(6,13)
          ast.Name(6,13)[a]
